### PR TITLE
bmi270: add CRT, compute CAS only once

### PR DESCRIPTION
**Pull request by bkleiner**

### CODE REVIEW

File: src/driver/spi_bmi270.c [hunk 1]
```diff
@@ -59,6 +69,49 @@ void bmi270_configure() {
...
+
+  bmi270_write(BMI270_REG_CMD, 0x02);
+  time_delay_ms(5000);
+
+  uint8_t crt_check = bmi270_read(BMI270_REG_GYR_CRT_CONF);
+  uint8_t crt_check_count = 0;
+  time_delay_ms(1000);
+
+  while ((crt_check != 0x00) && (crt_check_count++ < 5)) {
+    crt_check = bmi270_read(BMI270_REG_GYR_CRT_CONF);
```
Comment (DusKing1) [inline src/driver/spi_bmi270.c:98]: Please put a breakpoint at this line and check the read result for the `BMI270_REG_GYR_CRT_CONF`, should be `0x00` if CRT done.

File: src/driver/spi_bmi270.c [hunk 2]
```diff
@@ -59,6 +69,49 @@ void bmi270_configure() {
...
+    if (crt_check == 0x00) {
+      break;
+    }
+  }
+
+  bmi270_write(BMI270_REG_FEAT_PAGE, 0x00);
+  time_delay_ms(1);
+
+  const uint8_t gain_status = bmi270_read(BMI270_REG_FEATURES_0_GYR_GAIN_STATUS + 1);
+  if (gain_status == 0x0000) {
```
Comment (DusKing1) [inline src/driver/spi_bmi270.c:109]: This is the high byte so should be `0x00` instead of `0x0000`

File: src/driver/spi_bmi270.c (hunk 1 above)
Comment (bkleiner) [reply src/driver/spi_bmi270.c]: will do.

**Comment by DusKing1** [comment]
It's still worth mentioning that during the CRT the ACC is turned on and if any motion is detected the CRT will abort so it's pretty important to put the quad/drone on a stable surface to make sure it won't "move" during the CRT or say before the IMU fully "boot up".

**Comment by bkleiner** [comment]
i refactored the code to run CRT only after the gyro was detected sitting still, it is just too easy to move the craft after plugging in (especially with whoops).
with that ensured the check seems to come back with the expected zero values quite reliably.

**Review by DusKing1 — APPROVED** [review]

**Pull request merged**